Framework/Algorithms/src/EstimateResolutionDiffraction.cpp

```diff
@@ -225,100 +226,130 @@ void EstimateResolutionDiffraction::estimateDetectorResolution() {
+                                      const auto l2 = detectorInfo.l2(index.first);
+                                      const auto l_total = (l1 + l2);
+                                      realdet->shape();
+                                      const double dx = realdet->getWidth();
+                                      const double dy = realdet->getHeight();
+                                      const double dz = realdet->getDepth();
+                                      const double detdimSq = (dx * dx + dy * dy + dz * dz) * (0.5 * 0.5);
+                                      const double term = (detdimSq + sourceTerm) / (l_total * l_total);
+                                      return sum + term;
+                                    } else {
+                                      return sum;
+                                    }
+                                  }
+                                });
+      term2Sq = term2Sq / numDet;
     }
-    const double l_total = (l1 + l2);
-    const double term2Sq = (detdimSq + m_sourceDeltaLMetersSq) / (l_total * l_total);
 
-    // resolution in angle
-    const double twotheta = spectrumInfo.isMonitor(i) ? 0.0 : spectrumInfo.twoTheta(i);
-    const double theta = 0.5 * twotheta;
+    // resolution in angle - everything is in radians
+    const double theta = spectrumInfo.isMonitor(specNum) ? 0.0 : 0.5 * spectrumInfo.twoTheta(specNum);
 
-    double deltathetaSq = 0.; // this is in radians
+    double term3Sq;
     if (m_divergenceWS) {
-      deltathetaSq = m_divergenceWS->y(i)[0];
+      double deltathetaSq = m_divergenceWS->y(specNum)[0];
       deltathetaSq = deltathetaSq * deltathetaSq;
+      const double tan_theta = tan(theta);
+      term3Sq = (deltathetaSq + m_sourceDeltaThetaRadiansSq) / (tan_theta * tan_theta);
     } else {
-      auto &spectrumDefinition = spectrumInfo.spectrumDefinition(i);
-      const double solidangle =
-          std::accumulate(spectrumDefinition.cbegin(), spectrumDefinition.cend(), 0.,
-                          [&componentInfo, &detectorInfo, &samplepos](const auto sum, const auto &index) {
-                            if (!detectorInfo.isMasked(index.first)) {
-                              return sum + componentInfo.solidAngle(index.first, samplepos);
-                            } else {
-                              return sum;
-                            }
-                          });
-      deltathetaSq = solidangle;
+      const double sourceTerm = m_sourceDeltaThetaRadiansSq;
+      const double solidAngle = std::accumulate(
+          spectrumDefinition.cbegin(), spectrumDefinition.cend(), 0.,
+          [&componentInfo, &samplepos, &detectorInfo, &sourceTerm](const auto sum, const auto &index) {
+            if (detectorInfo.isMasked(index.first) || detectorInfo.isMonitor(index.first)) {
+              return sum;
+            } else {
+              const double theta = 0.5 * detectorInfo.twoTheta(index.first); // angle of the pixel
```

Comment:
How does this  theta differ from the theta in line 289?

Reply:
This is for the pixel where that was for the average over the spectrum